### PR TITLE
Plan form name field bug on edit

### LINKED ISSUE #1605: Plan Names Reported to be Changing in Thailand Replica3
**Describe the bug or issue**
CHAI have reported a case where plans names in replica3 changed when they got updated from draft to another status e.g. complete or retired. This mostly affects plans from 2019 to 2020. However, there are some plans also affected from 2021. An example on the changed names is on the screenshot below on the plan highlighted in red and green. The correct name for the highlighted plans should be `A1-ห้วยเกษม (7603020201)-สุพรรณ_สุกหอม-2021-05-11-Site`.

The ID for the complete plan is: `0fde2fb5-fbef-4786-ab28-eb9fb29bc9f0` , for the retired plan is: `788717f3-e530-440e-9f4c-ac76995c29f8`. A sample draft plan whose name value is correct is: `6ceb7df1-8b70-4fb5-be0d-b0b78e765d1a`.

It would be good to investigate this and identify if we can get to understand what is causing this.

image

**Expected behavior**
The plan names should remain unchanged.

**Additional context**

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #1605 
This pr ensures plan name and title fields are not edited when a user edits other fields

### CODE REVIEW

File: src/components/forms/PlanForm/helpers.ts [hunk 1]
```diff
@@ -880,8 +885,8 @@ export function getPlanFormValues(planObject: PlanDefinition): PlanFormFields {
     interventionType,
     jurisdictions: planObject.jurisdiction.map(e => ({
       id: e.code,
-      name: e.code,
-    })) /** a little cheating: since we dnt have the name yet, we just use code */,
+      name: '',
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/helpers.ts:888]: This makes the name by default `''`.  Why?

File: src/components/forms/PlanForm/tests/fixtures.ts [hunk 2]
```diff
@@ -936,7 +936,7 @@ export const planFormValues2 = {
   fiStatus: 'A2',
   identifier: '356b6b84-fc36-4389-a44a-2b038ed2f38d',
   interventionType: 'FI',
-  jurisdictions: [{ id: '3952', name: '3952' }],
+  jurisdictions: [{ id: '3952', name: '' }],
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/fixtures.ts:939]: Why is this change desirable?

File: src/components/forms/PlanForm/helpers.ts (hunk 1 above)
Comment (ciremusyoka) [reply src/components/forms/PlanForm/helpers.ts:888]: This helps avoid replacing of `name` or `title` with the code if a user tries to do edits before the  fetch of jurisdiction name (done by another [component](https://github.com/onaio/reveal-frontend/blob/master/src/containers/pages/InterventionPlan/UpdatePlan/PlanLocationNames/index.tsx)) is complete.

File: src/components/forms/PlanForm/tests/fixtures.ts (hunk 2 above)
Comment (ciremusyoka) [reply src/components/forms/PlanForm/tests/fixtures.ts:939]: Caused by the change you mentioned [above](https://github.com/onaio/reveal-frontend/pull/1612#discussion_r645735594)

**Review by moshthepitt — APPROVED** [review]

**Review by p-netm — APPROVED** [review]

**Pull request merged**